src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -0,0 +1,338 @@
+                )
+
+        attachments = Attachments(
+            manifests=manifest_properties_list,
+            fileSystem=session._job_attachment_details.job_attachments_file_system,
+        )
+
+        storage_profiles_path_mapping_rules_dict: dict[str, str] = {
+            str(rule.source_path): str(rule.destination_path)
+            for rule in session._job_details.path_mapping_rules
+        }
+
+        # Generate absolute Path Mapping to local session (no storage profile)
+        # returns root path to PathMappingRule mapping
+        dynamic_mapping_rules: dict[str, PathMappingRule] = (
+            session._asset_sync.generate_dynamic_path_mapping(
+                session_dir=session._session.working_directory,
+                attachments=attachments,
+            )
+        )
+
+        # Aggregate manifests (with step step dependency handling)
+        merged_manifests_by_root: dict[str, BaseAssetManifest] = (
+            session._asset_sync._aggregate_asset_root_manifests(
+                session_dir=session._session.working_directory,
+                s3_settings=s3_settings,
+                queue_id=session._queue_id,
+                job_id=session._queue._job_id,
+                attachments=attachments,
+                step_dependencies=step_dependencies,
+                dynamic_mapping_rules=dynamic_mapping_rules,
+                storage_profiles_path_mapping_rules=storage_profiles_path_mapping_rules_dict,
+            )
+        )
+
+        if self._start_vfs(
+            session=session,
+            attachments=attachments,
+            merged_manifests_by_root=merged_manifests_by_root,
+            s3_settings=s3_settings,
+        ):
+            # successfully launched VFS
+            return
+
+        job_attachment_path_mappings = list([asdict(r) for r in dynamic_mapping_rules.values()])
+
+        # Open Job Description session implementation details -- path mappings are sorted.
+        # bisect.insort only supports the 'key' arg in 3.10 or later, so
+        # we first extend the list and sort it afterwards.
+        if session._session._path_mapping_rules:
+            session._session._path_mapping_rules.extend(
+                OpenjdPathMapping.from_dict(r) for r in job_attachment_path_mappings
+            )
+        else:
+            session._session._path_mapping_rules = [
+                OpenjdPathMapping.from_dict(r) for r in job_attachment_path_mappings
+            ]
+
+        # Open Job Description Sessions sort the path mapping rules based on length of the parts make
+        # rules that are subsets of each other behave in a predictable manner. We must
+        # sort here since we're modifying that internal list appending to the list.
+        session._session._path_mapping_rules.sort(key=lambda rule: -len(rule.source_path.parts))
+
+        # =========================== TO BE DELETED ===========================
+        path_mapping_file_path: str = os.path.join(
```

Comment:
nit this will be deleted later right?

Reply:
Yes. In order to delete this, I'd need to do some additional change for the CLI to read a different format. The deletion will come once that's complete.